src/graph/on_disk_graph.cpp

```diff
@@ -70,6 +78,21 @@ OnDiskGraphScanStates::OnDiskGraphScanStates(ClientContext* context, std::span<R
     relIDVector =
         std::make_unique<ValueVector>(LogicalType::INTERNAL_ID(), context->getMemoryManager());
     relIDVector->state = state;
+    std::optional<column_id_t> edgePropertyID;
+    if (edgePropertyIndex) {
+        // Edge property scans are only supported for single table scans at the moment
+        KU_ASSERT(tables.size() == 1);
+        // TODO(bmwinger): If there are both a predicate and a custom edgePropertyIndex, they will
```

Comment:
You can probably leave this TODO to me or ziyi. I think the ultimate solution is to pass columnIDs or property expressions directly from the front-end. And we don't need to resolve the case of duplicate scanning.

Reply:
Okay